Makefile

```diff
@@ -114,19 +114,20 @@ dist/flannel dist/loopback dist/host-local:
 # .go-pkg-cache can't be used (since tests run as root)
 .PHONY: test-containerized
 ## Run the tests in a container (as root)
-test-containerized: run-etcd run-k8s-apiserver build-containerized dist/host-local
+test-containerized: build-containerized dist/host-local;
 	for cniversion in "0.2.0" "0.3.0" ; do \
-	docker run --rm --privileged --net=host \
-	-e ETCD_IP=$(LOCAL_IP_ENV) \
-	-e CNI_SPEC_VERSION=$cniversion \
-	-e LOCAL_USER_ID=0 \
-	-e PLUGIN=calico \
-	-v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw \
-	$(CALICO_BUILD) sh -c '\
-			cd  /go/src/github.com/projectcalico/cni-plugin && \
-			ginkgo'
-	make stop-etcd
-	end
+		make run-etcd; make run-k8s-apiserver; \
```

Comment:
I don't think this is the right approach. I think creating a new make target called something like test all versions would be better. If you don't agree or understand then lets talk f2f